attributes/attributes.go

```diff
@@ -121,13 +122,62 @@ func (a *Attributes) String() string {
 	return sb.String()
 }
 
-func str(x any) string {
-	if v, ok := x.(fmt.Stringer); ok {
-		return v.String()
+const nilAngleString = "<nil>"
+
+func str(x any) (s string) {
+	if x == nil { // NOTE: typed nils will not be caught by this check
+		return nilAngleString
```

Comment:
Why do we have to handle this case explicitly? Can't we rely on the `fmt` package for this as well? The `%#v` formatting directive seems to produce an output of the form `(type-name)(nil)` which seems good enough for us.